code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -176,36 +131,44 @@
 	icon_state = "esaw_0"
 	icon_state_on = "esaw_1"
 	hitcost = 75 //Costs more than a standard cyborg esword
-	item_color = null
 	w_class = WEIGHT_CLASS_NORMAL
 	sharpness = IS_SHARP
 	light_color = "#40ceff"
-	possible_colors = null
-
-/obj/item/weapon/melee/energy/sword/cyborg/saw/Initialize()
-	. = ..()
-	icon_state = "esaw_0"
-	item_color = null
 
-/obj/item/weapon/melee/energy/sword/cyborg/saw/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+/obj/item/weapon/melee/transforming/energy/sword/cyborg/saw/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
 	return 0
 
-/obj/item/weapon/melee/energy/sword/saber
+/obj/item/weapon/melee/transforming/energy/sword/saber
+	var/list/possible_colors = list("red" = LIGHT_COLOR_RED, "blue" = LIGHT_COLOR_GREEN, "green" = LIGHT_COLOR_LIGHT_CYAN, "purple" = LIGHT_COLOR_LAVENDER)
+	var/hacked = FALSE
+
+/obj/item/weapon/melee/transforming/energy/sword/saber/Initialize(mapload)
+	. = ..()
+	if(LAZYLEN(possible_colors))
+		var/set_color = pick(possible_colors)
+		item_color = set_color
+		light_color = possible_colors[set_color]
 
-/obj/item/weapon/melee/energy/sword/saber/blue
+/obj/item/weapon/melee/transforming/energy/sword/saber/process()
+	. = ..()
+	if(hacked)
+		var/set_color = pick(possible_colors)
+		light_color = possible_colors[set_color]
+		update_light()
+
+/obj/item/weapon/melee/transforming/energy/sword/saber/blue
 	possible_colors = list("blue")
 
-/obj/item/weapon/melee/energy/sword/saber/purple
+/obj/item/weapon/melee/transforming/energy/sword/saber/purple
 	possible_colors = list("purple")

```

Comment:
Forgot to update the other `possible_colors`. Might be a good idea to make the assoc a define on this file, but doenst matter that much.